### ISSUE TITLE
GenerateKubeconfigActionHandler panics if LocalClusterAuthEndpoint is nil

### ISSUE THREAD

**Issue by dramich** [rancher/rancher#23783]
```
2019/10/30 23:20:47 [ERROR] Panic serving api request:
goroutine 879946 [running]:
runtime/debug.Stack(0xc037ea5000, 0x2d2e580, 0x5d5de80)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x39e2a00, 0xc046484900)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:175 +0x7a
panic(0x2d2e580, 0x5d5de80)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/rancher/rancher/pkg/api/customization/cluster.ActionHandler.GenerateKubeconfigActionHandler(0x7ffac55c82f8, 0xc0003ef8c0, 0x3a64720, 0xc000acab40, 0x7ffac55c8318, 0xc0003ef8c0, 0x3a34d00, 0xc00017b380, 0xc000857600, 0x3a64ea0, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/cluster/actions_kubeconfig.go:24 +0xbd
github.com/rancher/rancher/pkg/api/customization/cluster.ActionHandler.ClusterActionHandler(0x7ffac55c82f8, 0xc0003ef8c0, 0x3a64720, 0xc000acab40, 0x7ffac55c8318, 0xc0003ef8c0, 0x3a34d00, 0xc00017b380, 0xc000857600, 0x3a64ea0, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/cluster/actionhandler.go:45 +0x107f
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.handleAction(0xc0172ff720, 0xc09df06120, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:263 +0x62
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).handle(0xc00084d3f0, 0x39e2a00, 0xc046484900, 0xc072902800, 0xc072902800, 0xc00fe79900, 0xc08c9774a0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:251 +0xd2
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP(0xc00084d3f0, 0x39e2a00, 0xc046484900, 0xc072902800)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:180 +0x7d
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc00126a840, 0x39e2a00, 0xc046484900, 0xc072902200)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:212 +0xe3
github.com/rancher/rancher/pkg/auth/requests.authHeaderHandler.ServeHTTP(0x39b1ec0, 0xc00149e780, 0x398a960, 0xc00126a840, 0x39c1ac0, 0xc000857600, 0x39b1e80, 0xc0014b2870, 0x39e2a00, 0xc046484900, ...)
	/go/src/github.com/rancher/rancher/pkg/auth/requests/filter.go:110 +0x72b
github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP(0x3989b80, 0xc0013baf80, 0x39e2a00, 0xc046484900, 0xc072902200)
	/go/src/github.com/rancher/rancher/pkg/websocket/handler.go:35 +0xa1
github.com/rancher/rancher/pkg/audit.auditHandler.ServeHTTP(0x3989d60, 0xc0003e3fe0, 0x0, 0x39e2a00, 0xc046484900, 0xc072902200)
	/go/src/github.com/rancher/rancher/pkg/audit/filter.go:31 +0x561
github.com/rancher/rancher/server/responsewriter.ContentType.func1(0x39d0340, 0xc08c9773b0, 0xc072902200)
	/go/src/github.com/rancher/rancher/server/responsewriter/content.go:33 +0x93
net/http.HandlerFunc.ServeHTTP(0xc000d1e240, 0x39d0340, 0xc08c9773b0, 0xc072902200)
	/usr/local/go/src/net/http/server.go:1995 +0x44
github.com/rancher/rancher/server/responsewriter.Gzip.func1(0x39e2780, 0xc0c78b5960, 0xc072902200)
	/go/src/github.com/rancher/rancher/server/responsewriter/gzip.go:60 +0x1b6
net/http.HandlerFunc.ServeHTTP(0xc000d1e260, 0x39e2780, 0xc0c78b5960, 0xc072902200)
	/usr/local/go/src/net/http/server.go:1995 +0x44
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc00126a780, 0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:212 +0xe3
github.com/rancher/rancher/pkg/dynamiclistener.httpRedirect.func1(0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/go/src/github.com/rancher/rancher/pkg/dynamiclistener/server.go:506 +0x259
net/http.HandlerFunc.ServeHTTP(0xc0009fb020, 0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/usr/local/go/src/net/http/server.go:1995 +0x44
github.com/rancher/rancher/pkg/dynamiclistener.(*Server).cacheIPHandler.func1(0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/go/src/github.com/rancher/rancher/pkg/dynamiclistener/server.go:436 +0x101
net/http.HandlerFunc.ServeHTTP(0xc0009fb060, 0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/usr/local/go/src/net/http/server.go:1995 +0x44
net/http.serverHandler.ServeHTTP(0xc0015fc750, 0x39e2780, 0xc0c78b5960, 0xc076d9b000)
	/usr/local/go/src/net/http/server.go:2774 +0xa8
net/http.(*conn).serve(0xc0405b4280, 0x3a032c0, 0xc12e885880)
	/usr/local/go/src/net/http/server.go:1878 +0x851
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2884 +0x2f4
```

**Comment by iam-merlin** [comment]
This issue can be reproduced when you edit your cluster (LocalClusterAuthEndpoint will not be set)

**Comment by sangeethah** [comment]
Regression cases - Verify kubeconfig files come out as expected for clusters with and without the cluster auth endpoint and that we are able to connect to the cluster as using the generated kubeconfig files.

**Comment by rmweir** [comment]
From PR https://github.com/rancher/rancher/pull/23912:

> This happens because to use access.byId the client cluster struct needs to be used, opposed to the regular management struct. Typically there is a default localClusterAuthEndpoint but all struct fields on the client cluster struct are pointers, and so the default is nil. This is a generated struct.
> 
> This only occurs when the request lacks a localClusterAuthEndpoint or it is explicitly set to null in the request. The UI is currently omitting the localClusterAuthEndpoint (see following issue):
> #23913

**Comment by rmweir** [comment]
Testing steps:
1. create import cluster
2. edit imported cluster, change description, save
3. select "Kubeconfig File" or use generateKubeConfig action in API UI

**Comment by jiaqiluo** [comment]
The bug is reproduced in v2.3.2

The steps are the same as the above comment:

- create import cluster
- edit imported cluster, change the description, save
- select "Kubeconfig File" or use generateKubeConfig action in API UI

Result:
- panic errors in the log

**Comment by jiaqiluo** [comment]
The bug fix is validated in 2.3-head 5187fce31

- create import cluster
- select "Kubeconfig File" or use generateKubeConfig action in API UI ==> The kubeconfig is generated and no errors in the log
- edit imported cluster, change the description, save
- select "Kubeconfig File" or use generateKubeConfig action in API UI ==> The kubeconfig is generated and no errors in the log
